scripts/ci/gitlab/timestamp.yml

```diff
@@ -0,0 +1,27 @@
+.timestamp:
+  before_script:
+    - |
+      echo "Install timestamp handler"
+      eval '
+        # https://gist.github.com/altaua/477285db40181ef1793bbed5bae32c6b
+        # based on https://gist.github.com/jstine35/e0fc0e06ec06d74bc3ebd67585bf2a1d
+
+        s_datestamp() {
+          TZ=UTC
+          eof=""
+          while [[ -z "${eof}" ]]; do
+            IFS="" read -r line || eof=1
+            printf "[%(%F %T)T] %s\n" -1 "${line}"
+          done
+        }
+
+        cleanup() {
+          exec >/dev/null 2>&1
+          wait "${TIMESTAMP_PID}"
+        }
+
+        exec > >(s_datestamp) 2>&1
+
+        TIMESTAMP_PID=$!
+        trap cleanup EXIT
```

Comment:
Interesting. I don't understand what is it for) Could you please explain?

Reply:
Oh, that's to fix a bug in the original script: The main job script might finish before the s_datestamp process substitution has had a chance to process all output; if we don't wait for it, bash will just exit immediately, killing s_datestamp and thus discarding the last chunk of output.

Reply:
(Incidentally, that's what made us not see the timeout error message in test-linux-stable-int for #13047)